code/game/objects/items/weapons/circuitboards/circuitboard.dm

```diff
@@ -38,3 +38,16 @@
 	if (istype(M, build_path))
 		return TRUE
 	return FALSE
+
+/obj/item/weapon/circuitboard/examine(user, distance)
+	. = ..()
+	// gets the required components and displays it in a list to the user when examined.
+	if(length(req_components))
+		var/list/listed_components = list()
+		for(var/requirement in req_components)
+			var/atom/placeholder = requirement
+			if(!ispath(placeholder))
+				continue
+			listed_components += list("[req_components[placeholder]] [initial(placeholder.name)]")
+			var/verbage = english_list(listed_components)
+			to_chat(user, SPAN_NOTICE("<span class='notice'>Required components: [verbage].</span>"))
```

Comment:
Would it spams the chat if `			var/verbage = english_list(listed_components)
			to_chat(user, SPAN_NOTICE("<span class='notice'>Required components: [verbage].</span>")) ` is inside in the loop ?

Reply:
Good catch!